packages/server/server.py

```
@@ -0,0 +1,249 @@
+        return flask.jsonify(error=True, message="invalid url"), 400
+    
+    try:
+        info = downloader.download.info(url)
+        video_id = info["id"]
+        
+        # Check if already downloaded
+        already_downloaded, existing_file = is_already_downloaded(video_id)
+        if already_downloaded:
+            # Create progress file showing completion
+            os.makedirs(OUTPUTS_DIR / "info", exist_ok=True)
+            with open(OUTPUTS_DIR / "info" / f'{video_id}.json', 'w') as wo:
```

Ruff (0.12.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)